server/src/main/java/org/elasticsearch/cluster/coordination/Publication.java

```diff
@@ -333,9 +333,15 @@ public void onResponse(PublishWithJoinResponse response) {
 
                 assert state == PublicationTargetState.SENT_PUBLISH_REQUEST : state + " -> " + PublicationTargetState.WAITING_FOR_QUORUM;
                 state = PublicationTargetState.WAITING_FOR_QUORUM;
-                handlePublishResponse(response.getPublishResponse());
 
-                assert publicationCompletedIffAllTargetsInactiveOrCancelled();
+                try {
+                    handlePublishResponse(response.getPublishResponse());
```

Comment:
should we move this exception handling to the `handlePublishResponse` method? It should only wrap the `Publication.this.handlePublishResponse` part

Reply:
Sure, see b38833df1a8327a38c341163c46a0d525b39bdca.